src/framework/mlt_consumer.c

```diff
@@ -712,8 +712,9 @@ mlt_frame mlt_consumer_get_frame( mlt_consumer self )
 		if ( test_card != NULL )
 			mlt_properties_set_data( frame_properties, "test_card_producer", test_card, 0, NULL, NULL );
 
-		// Pass along the interpolation and deinterlace options
+		// Pass along the channel convert, interpolation and deinterlace options
 		// TODO: get rid of consumer_deinterlace and use profile.progressive
+		mlt_properties_set( frame_properties, "audiochannel.convert", mlt_properties_get( properties, "channelconvert" ) );
```

Comment:
I chose "channelconvert " as the consumer property name for specifying "mix" or "dupdrop" modes.

Any suggestions for a different name?
Should this new consumer property be documented somewhere?

Reply:
There is a convention that consumer properties propagated from the consumer to the frame begin with "consumer_" (consumer_deinterlace, consumer_tff, etc.) although I see how you are targeting the corresponding filter with your name for the frame property. That is sort of an inversion of the convention, which has been to publish properties and let whatever filter use them. If you are going to target this filter, then use the exact name "audiochannels." Since, we use the plural form "channels" everywhere else including consumer and frame properties, I prefer "channels_convert" as the base property name.

Reply:
Hmm, do we need to propagate this property in mlt_tractor.c:producer_get_audio()?

Reply:
I had in mind that any filter could intercept this property and use/apply it. So I don't think it has to specifically target the audiochannels filter. Or example, we could create a new filter that uses avresample and that filter should also use this property. I will change it to "consumer_channels_convert"